searchcore/src/vespa/searchcore/proton/matching/querylimiter.cpp

```diff
@@ -25,14 +25,14 @@ QueryLimiter::grabToken(const Doom & doom)
             _cond.wait_for(guard, left);
         }
     }
-    _activeThreads++;
+    _activeThreads = _activeThreads + 1;
 }
 
 void
 QueryLimiter::releaseToken()
 {
     std::lock_guard<std::mutex> guard(_lock);
-    _activeThreads--;
+    _activeThreads = _activeThreads - 1;
```

Comment:
I think that it is better to remove volatile instead. It seems all use is while holding a lock.